cns/restserver/restserver.go

```diff
@@ -34,6 +34,28 @@ var (
 	ncVersionURLs sync.Map
 )
 
+type lockedDurationSet struct {
+	sync.Mutex
+	waits map[string]time.Time
+}
+
+func (m *lockedDurationSet) Push(key string) {
+	m.Lock()
```

Comment:
Why do we need separate lock? Can we update this while acquiring the httpRestService lock only?

Reply:
then we increase lock contention, and block unrelated things trying to access other synchronized state. I think it's better to scope the mutex tightly to the datastructure that it is protecting access to.